cpplint.py

```diff
@@ -1318,8 +1318,10 @@ def RepositoryName(self):
       # searching up from the current path.
       root_dir = current_dir = os.path.dirname(fullname)
       while current_dir != os.path.dirname(current_dir):
-        if (os.path.exists(os.path.join(current_dir, ".git")) or
-            os.path.exists(os.path.join(current_dir, ".hg")) or
+        if (os.path.exists(os.path.join(current_dir, ".git"))):
+          root_dir = current_dir
+          break  # stop here, we might be in a submodule
+        if (os.path.exists(os.path.join(current_dir, ".hg")) or
```

Comment:
the same argument would apply to Mercurial subrepositories, no?

Reply:
I haven't used Mercurial for years, so I would rather someone knowledgeable could check.

Reply:
Not a reason not to merge anyway.